pyproject.toml

```diff
@@ -53,6 +53,45 @@ boto3 = [
 Repository = "https://github.com/aio-libs/aiobotocore"
 Documentation = "https://aiobotocore.aio-libs.org"
 
+[dependency-groups]
+botocore-dev = [
+    # keep in sync with https://github.com/boto/botocore/blob/develop/requirements-dev.txt
+    # "wheel==0.43.0",
+    # "behave==1.2.5",
+    # "jsonschema==4.21.1",
+    "coverage==7.2.7",
+    # "setuptools==71.1.0;python_version>='3.12'",
+    # "packaging==24.1;python_version>='3.12'", # Requirement for setuptools>=71
+
+    # Pytest specific deps
+    "pytest==8.1.1",
+    "pytest-cov==5.0.0",
+    "pytest-xdist==3.5.0",
+    # "atomicwrites>=1.0", # Windows requirement
+    # "colorama>0.3.0", # Windows requirement
+]
+dev = [
+    "dill ~=0.3.3", # Used in test_patches.py
+    "docker ~=7.1",
+    "docutils ~=0.16", # Used in test_version.py
+    "moto[server,s3,sqs,awslambda,dynamodb,cloudformation,sns,batch,ec2,rds] ~=4.2.9",
+    "packaging ~=24.1", # Used in test_version.py
+    "pip ~=24.3.1", # Used in test_version.py
+    "pre-commit ~=3.5.0",
+    "pytest-asyncio ~=0.23.8",
+    "requests ~=2.32.3", # Used in test_version.py
+    "time-machine ~=2.15.0",
+    "tomli; python_version<'3.11'", # Used in test_version.py
+    "werkzeug ~=3.0.6", # Used in moto_server.py
```

Comment:
this is a side-effect of moto, if moto changes werkzeug we need to change our impl as well.  IOW this is not our choice

Reply:
if it makes you feel better can import this from moto directly ;)

Reply:
Sorry, if I'm mistaken, but I thought we had resolved that as part of #1246. We use `werkzeug` directly in unit tests and that specification guards us from regressions should moto drop that dependency. I agree that we might then subsequently also drop werkzeug, but that's a separate issue, IMO.

Reply:
Exactly: https://github.com/aio-libs/aiobotocore/blob/7432338/tests/moto_server.py#L12C1-L12C16

Reply:
no, see my reply: https://github.com/aio-libs/aiobotocore/pull/1246#discussion_r1906210969

Reply:
if moto stops using that library, we NEED to update our code, and having the import fail is a feature letting us know something needs to be re-written to match moto

Reply:
if we include that library we're going to have some weird side-effect behavior of moto moves away from using the library

Reply:
It's a direct dependency right now: https://github.com/aio-libs/aiobotocore/pull/1246#discussion_r1906234448. Pretending that it isn't wouldn't change the fact that it is, it'd only make it more confusing.

Reply:
following rules for rules sake is fools errand, there will always be exceptions

Reply:
In the end, it should be @thehesiod who makes the call what qualifies as a dev dependency. I was working under the wrong assumption that we had reached consensus and that's why I jumped the gun in #1246. If you ask for it, I will revert dev dependencies to the previous state.

Reply:
how about this, we pull werkzerg import from moto?  then it's not a direct dependency

Reply:
> how about this, we pull werkzerg import from moto? then it's not a direct dependency

I can live with that :-)

Reply:
cool once that's done I'll approve

Reply:
Well, we cannot switch to importing indirectly from moto, because:
1. `moto` uses the `from werkzeug.foo import bar` pattern.
2. In `moto_server.py`, we use `werkzeug.serving.generate_adhoc_ssl_context()`.
3. That function is not used in `moto==4.2.14`, i.e. that function does not become part of the moto namespace.

How does that change you assessment?